src/utils/dom-to-image.js

```diff
@@ -342,16 +344,20 @@ function makeSvgDataUri(node, width, height) {
   return Promise.resolve(node)
     .then(nd => {
       nd.setAttribute('xmlns', 'http://www.w3.org/1999/xhtml');
-      return new window.XMLSerializer().serializeToString(nd);
-    })
-    .then(util.escapeXhtml)
-    .then(xhtml =>
-      `<foreignObject x="0" y="0" width="100%" height="100%">${xhtml}</foreignObject>`
-    )
-    .then(foreignObject =>
-      `<svg xmlns="http://www.w3.org/2000/svg" width="${width}" height="${height}">${foreignObject}</svg>`
-    )
-    .then(svg => `data:image/svg+xml;charset=utf-8,${svg}`);
+      const serializedString =  new window.XMLSerializer().serializeToString(nd);
+
+      const xhtml = util.escapeXhtml(serializedString);
+      const foreignObject = `<foreignObject x="0" y="0" width="100%" height="100%">${xhtml}</foreignObject>`;
+      const svgStr = `<svg xmlns="http://www.w3.org/2000/svg" width="${width}" height="${height}">${foreignObject}</svg>`;
+
+      // Optimizing SVGs in data URIs
+      // see https://codepen.io/tigt/post/optimizing-svgs-in-data-uris
+      // the best way of encoding SVG in a data: URI is data:image/svg+xml,[actual data].
+      // We don’t need the ;charset=utf-8 parameter because the given SVG is ASCII.
+      const dataUri = svgToMiniDataURI(svgStr);
```

Comment:
you can simply return the result of the __svgToMiniDataURI__

Reply:
done